src/main/java/backend/RepoID.java

```diff
@@ -0,0 +1,68 @@
+package backend;
+
+import util.Utility;
+
+import java.util.Optional;
+
+/**
+ * Represents the ID of a repository. To be used to ensure safe comparison
+ */
+public final class RepoID {
+
+    private final String repoIDString;
```

Comment:
We can compute this from the other two fields. Maybe we could use a getter that appends them instead of storing this separately?